securedrop_client/storage.py

```diff
@@ -532,12 +521,14 @@ def delete_single_submission_or_reply_on_disk(obj_db: Union[File, Message, Reply
             logging.info('File %s already deleted, skipping', file_to_delete)
 
 
-def rename_file(data_dir: str, filename: str, new_filename: str) -> None:
-    filename, _ = os.path.splitext(filename)
-    new_filename, _ = os.path.splitext(new_filename)
+def rename_file(data_dir: str, file: Union[Message, File, Reply], new_filename: str) -> None:
```

Comment:
this method should not be needed anymore right? since we now only store the original filename on disk we don't need to ever rename that file if the journalist designation changes. we also don't store replies or messages on disk.

Reply:
When a file is downloaded, it's named using the server filename until it's decrypted, at which point `File.filename` and the file on disk should have the name from the gzip header. If decryption fails, the file might be on disk with a name based on the designation. If we don't worry about that, then if the designation and server filenames change, the content will just be downloaded again, and we'll have an extra file in the File's directory. Maybe we're going to need some housekeeping down the road anyway, but for now, I was going to preserve this.

Reply:
ah makes sense to preserve that for now, and we may want to rethink keeping files around that fail to decrypt